addons/hr_expense/models/hr_expense.py

```diff
@@ -311,6 +315,23 @@ def _check_product_uom_category(self):
                     expense=expense.name, product=expense.product_id.name,
                 ))
 
+    @api.constrains('account_id', 'currency_id', 'payment_mode')
+    def _check_account_currency_consistency(self):
+        for expense in self:
+            account_currency = expense.account_id.currency_id
+            if account_currency and account_currency != expense.company_currency_id:
+                if account_currency != expense.currency_id:
+                    raise ValidationError(_(
```

Comment:
```suggestion
            account_currency = expense.account_id.currency_id
            if account_currency != expense.company_currency_id:
                if account_currency != expense.currency_id:
                    raise ValidationError(_(
```
company_currency_id is not stored, and related to the company currency required field. So it should never be False